Code cleaning

Brett208 opened the pull request:
Since I was spending some time in OP2Helper, I wanted to clean / modernize some of the functions.

@DanRStevens, if any of it seems controversial, we can just exclude it. 

I mostly stayed out of BaseBuilderV2.

-Brett

Brett208 committed:
Add brackets to single line statements

Brett208 committed:
Standardize for statements

Brett208 committed:
Remove forward declares of variables in functions

Brett208 committed:
Rename trig to trigger

Brett208 committed:
Clean function Create CreateInitialCombatUnits

DanRStevens commented on OP2Helper.cpp:
```diff
@@ -54,15 +54,21 @@ void CreateTubeOrWallLine(int x1, int y1, int x2, int y2, map_id type)
+	}
+	if (y1 > y2) { 
+		y1 ^= y2; y2 ^= y1; y1 ^= y2;
+	}
```
Ahh yes, I was looking at this a few days ago and thought this code really needed to go. It's using the "clever" XOR swap algorithm, which is really not very readable, and not a good idea to use inline in code like this. It should be replaced by something like `std::swap(x1, x2)` and `std::swap(y1, y2)`. See [`std::swap`](https://en.cppreference.com/w/cpp/algorithm/swap). It seems there is a choice of header includes for `swap`.

Brett208 replied:
But how am I supposed to practice bit twiddling now???

DanRStevens commented on BaseBuilder.cpp:
```diff
@@ -83,14 +83,13 @@ void CreateBeacons(int numBeacons, const BeaconInfo beacon[])
+	Unit unit;
+	for (int i = TethysGame::InitialUnits(); i > 0; --i)
 	{
-		Unit unit;
```
Why was `Unit` moved outside the for loop here? This looks like an accidental move.

DanRStevens reviewed (approved):
Thank you for the updates. Looks like good changes. Marked two places for possible further improvement.

----

We may want to consider just scrapping BaseBuilderV2 at some point. I don't think anyone is really using it, and it was never really completed. It was supposed to offer more readable syntax, though I'm not sure if it's worth it.

DanRStevens committed:
Fix improper forward declaration

In standard C++, only `enum`s with base types can be forward declared.
In particular, `enum class` enums may be forward declared.

For old style `enum`, we should instead just include the full definition.

DanRStevens committed:
Replace XOR swap with `std::swap`

This is vastly more readable.

DanRStevens committed:
Move `Unit` declaration into `for` loop

Brett208 commented on BaseBuilder.cpp:
```diff
@@ -86,9 +86,9 @@ void CreateInitialCombatUnits(int playerNum, int x, int y)
-	Unit unit;
 	for (int i = TethysGame::InitialUnits(); i > 0; --i)
 	{
+		Unit unit;
```
I was waffling about where this should be. Happy either way.

Brett208 reviewed (commented):
I was thinking maybe we could move BaseBuilderV2 to a branch since it isn't fully implemented yet. I guess if the branch languishes long enough it could be deleted eventually.

Thanks for implementing swap and carrying the cleaning a bit further.

DanRStevens commented:
I think you're right about moving the BaseBuilderV2 code to a branch.
